File: 04-Socket-UDP/esempi/01.01-unicast-java/udp-sender.js
/**
 * udp-sender.js
 * Esempio di mittente UDP in JavaScript/Node.js
 * Invia un messaggio UDP a un ricevitore
 * Versione: 1.0
 * Data: 25/01/26
 * Autore: Filippo Bilardo
 *
 * node udp-sender.js
 */

const dgram = require('dgram');

// Crea socket UDP
const socket = dgram.createSocket('udp4');

// Configurazione
const message = 'Hello, UDP!';
const host = 'localhost';
const port = 9876;

// Prepara il buffer
const buffer = Buffer.from(message);

// Invia il messaggio
socket.send(buffer, 0, buffer.length, port, host, (err) => {
    if (err) {
        console.error('Errore durante l\'invio:', err);
        socket.close();
        process.exit(1);
    }
    
    console.log('Messaggio inviato!');
    socket.close();
});

// Gestione errori
socket.on('error', (err) => {
    console.error('Errore socket:', err);
    socket.close();
    process.exit(1);
});

File: 04-Socket-UDP/esempi/01.01-unicast-java/UdpSender.java
/**
 * UdpSender.java
 * Esempio di mittente UDP in Java
 * Invia un messaggio UDP a un ricevitore
 * Versione: 1.0
 * Data: 18/10/25
 * Autore: Filippo Bilardo
 *
 * javac UdpSender.java && java UdpSender
 */
import java.net.DatagramPacket;
import java.net.DatagramSocket;
import java.net.InetAddress;

public class UdpSender {
    public static void main(String[] args) {
        try {
            DatagramSocket socket = new DatagramSocket();
            String message = "Hello, UDP!";
            byte[] buffer = message.getBytes();
            InetAddress address = InetAddress.getByName("localhost");

            DatagramPacket packet = new DatagramPacket(buffer, buffer.length, address, 9876);
            socket.send(packet);
            System.out.println("Messaggio inviato!");
            socket.close();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}